plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -273,19 +436,18 @@ public void testJoinPushdown()
         }
     }
 
-    private void assertConditionallyPushedDown(
+    private QueryAssert assertConditionallyPushedDown(
```

Comment:
make it void again. It's too late to add `ordered` on the returned.

you can add `ordered` parameter here

or maybe we can decompose this:

```
private static QueryAssert assertConditionallyPushedDown(
        QueryAssert queryAssert,
        boolean condition,
        PlanMatchPattern otherwiseExpected)
{
    if (condition) {
        return queryAssert.isFullyPushedDown();
    }

    return queryAssert.isNotFullyPushedDown(otherwiseExpected);
}
```

Reply:
```java
            assertConditionallyPushedDown(
                    assertThat(query(session, "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_string ASC LIMIT 2")).ordered(),
                    expectTopNPushdown,
                    topNOverTableScan)
```

vs

```java
            assertConditionallyPushedDown(
                    session,
                    "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_string ASC LIMIT 2",
                    true,  // boolean - ordered
                    expectTopNPushdown, // boolean
                    topNOverTableScan);
```

I guess composition it is.